src/dectris/neggia/data/PathResolverV2.cpp

```diff
@@ -0,0 +1,114 @@
+#include "PathResolverV2.h"
+#include <assert.h>
+#include <memory>
+#include "H5BTreeVersion2.h"
+#include "H5FractalHeap.h"
+#include "H5LocalHeap.h"
+#include "constants.h"
+
+#include <iostream>
+
+PathResolverV2::PathResolverV2(const H5ObjectHeader& root) : _root(root) {}
+
+ResolvedPath PathResolverV2::resolve(const H5Path& path) {
+    return resolvePathInHeader(_root, path);
+}
+
+ResolvedPath PathResolverV2::resolvePathInHeader(const H5ObjectHeader& in,
+                                                 const H5Path& path) {
+    H5ObjectHeader parentEntry(path.isAbsolute() ? _root : in);
+    for (auto itemIterator = path.begin(); itemIterator != path.end();
+         ++itemIterator)
+    {
+        auto item = *itemIterator;
```

Comment:
You could write `const auto item = *itemIterator++;` here. Then you don't need to increment in three other places.